azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/NetworkInterface.java

```diff
@@ -70,7 +70,7 @@
      *
      * @return public ip resource id
      */
-    String publicIpAddressId();
+    String primaryPublicIpAddressId();
```

Comment:
Fine will remove this public Ip Id. As you mentioned perf was the reason to introduce this property.